src/ORM/Association/BelongsToMany.php

```diff
@@ -581,24 +615,50 @@ public function isOwningSide(Table $side)
     }
 
     /**
+     * Sets the strategy that should be used for saving.
+     *
+     * @param string $strategy the strategy name to be used
+     * @throws \InvalidArgumentException if an invalid strategy name is passed
+     * @return $this
+     */
+    public function setSaveStrategy($strategy)
+    {
+        if (!in_array($strategy, [self::SAVE_APPEND, self::SAVE_REPLACE])) {
+            $msg = sprintf('Invalid save strategy "%s"', $strategy);
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->_saveStrategy = $strategy;
+
+        return $this;
+    }
+
+    /**
+     * Gets the strategy that should be used for saving.
+     *
+     * @return string the strategy to be used for saving
+     */
+    public function getSaveStrategy()
+    {
+        return $this->_saveStrategy;
+    }
+
+    /**
      * Sets the strategy that should be used for saving. If called with no
      * arguments, it will return the currently configured strategy
      *
+     * @deprecated Use setSaveStrategy()/getSaveStrategy() instead.
```

Comment:
We could agree to not include the (works until) version number anymore, as it is optional.

https://phpdoc.org/docs/latest/references/phpdoc/tags/deprecated.html

I guess it confuses people anyway, as some would expect to see since when something is deprecated instead untill it is kept in.

Adhering to SemVer mostly solves this problem.
You know if something is deprecated in 3.3.4, you will be able to use it untill the last 3.x version and you know it will be removed in 4.0.0.

But I am aware that sometimes deprecated stuff does not get removed right away, as maintainers forget about it, so keeping the information up to which version you officially can use it, could still be useful.